### PR TITLE
feat(makefile): migrate away from hardcoded binary

### PR DESCRIPTION
**Pull request by georgettica**
Upstream change that is relevant to this https://github.com/kubernetes-sigs/kubebuilder/pull/2313

**Description of the change:**
Make `operator-sdk` binary location be dynamic (allow multiple versions of operator-sdk /  containerized operator-sdk)

**Motivation for the change:**
allow working on the `Makefile` as a given file and modify an additional file

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [x] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Comment by georgettica** [comment]
maybe not all of the relevant docs have been updated but please help me understand what I need to modify if any

**Review by camilamacedo86 — CHANGES_REQUESTED** [review]
You need to run make generate to update all samples. 
Also, I would suggest rebasing with master

**Comment by georgettica** [comment]
rebsed and ran `make generate`
I am getting the error:
```
INFO[0001] creating the bundle
ERRO[0001] error creating the bundle: unable to find the content to be replaced
exit status 1
```

not sure if this is a real error ( I uninstalled regular operator-sdk binary I have so the new one will take precedence)

**Comment by camilamacedo86** [comment]
You must run make install to install the source code locally
then, you can use the bin installed locally to check your changes
by running make generate after that you will use your changes/bin from the source to re-generate all samples. If that does not work then, that means that your changes are breaking it.

**Comment by georgettica** [comment]
I'll try it again

**Review by jmrodri — APPROVED** [review]
/lgtm

File: changelog/fragments/01-move-away-from-operator-sdk-binary.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# release notes and/or the migration guide
+entries:
+  - description: >
+      Add a Parameter to the Makefile to allow modifying the
+      'operator-sdk' binary before running any commands.
+      This allows you to run `OPERATOR_SDK=~/local/operator-sdk-1.2 make bundle`
+      without modifying and of the default makefile provided
+    kind: addition
+
+    breaking: false
```
Comment (jmrodri) [inline changelog/fragments/01-move-away-from-operator-sdk-binary.yaml:11]: @camilamacedo86 do you think we need a migration step for this change? It definitely isn't required to work but not sure. Thoughts?

Comment (georgettica) [reply changelog/fragments/01-move-away-from-operator-sdk-binary.yaml:11]: @jmrodri seems this didn't get any ❤️ 
I will unrelated move to rebase this

**Comment by georgettica** [comment]
failing to `make generate` because of 
```shell
$ make generate
XXX
go run ./hack/generate/cncf-maintainers/main.go
go run ./hack/generate/cli-doc/gen-cli-doc.go
go run ./hack/generate/samples/generate_testdata.go
INFO[0000] writing sample directories under XXX/operator-sdk/testdata
INFO[0000] creating Helm Memcached Sample
INFO[0000] destroying directory for memcached helm samples
running: docker rmi -f quay.io/example/memcached-operator:v0.0.1
INFO[0000] creating directory
cleaning up tools
preparing testing directory: XXX/testdata/helm/memcached-operator
INFO[0000] setting domain and GVK
INFO[0000] creating the project using the helm chart in: (../../../hack/generate/samples/internal/helm/testdata/memcached-0.0.1.tgz)
running: operator-sdk init --plugins helm --domain example.com --group cache --version v1alpha1 --kind Memcached --helm-chart ../../../hack/generate/samples/internal/helm/testdata/memcached-0.0.1.tgz
INFO[0000] customizing the sample
INFO[0000] enabling prometheus metrics
INFO[0000] adding customized roles
INFO[0000] creating the bundle
ERRO[0000] error creating the bundle: unable to find the content to be replaced
exit status 1
make: *** [generate] Error 1
```

**Pull request closed without merging**